### PR TITLE
Add support for Quilt3

### PR DESCRIPTION
**Pull request by jinyeom**
## Overview
This PR adds support for Quilt 3.

## Todo
- [x] Test this with a dataset

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -18,6 +18,7 @@
         'pillow>=5.4.1',
 
         'quilt==2.9.15',
+        'quilt3==3.1.0',
```
Comment (jalabort) [inline setup.py:21]: Ah! That's nice! So we can keep using both versions of `quilt` until we completely transition to `version 3`?

File: nucleus/dataset/tools/quilt.py [hunk 2]
```diff
@@ -136,3 +172,36 @@ def update_pkg(
...
+    import quilt3
+
+    df.to_parquet(parquet)
+    
+    pkg_path = f'{user}/{package}'
+    pkg = quilt3.Package.browse(pkg_path, registry=registry)
+    pkg.set(parquet)
+
+    quilt3.login()
+    pkg.push(pkg_path)
```
Comment (jalabort) [inline nucleus/dataset/tools/quilt.py]: Nitpick: blank line

**Review by jalabort — APPROVED** [review]
LGTM. Very excited about this. Let me know how you get on with the design of the detection dataset consumer class.

File: setup.py (hunk 1 above)
Comment (jinyeom) [reply setup.py:21]: Yep!

**Pull request merged**